R/connect.R

```diff
@@ -50,17 +50,17 @@ NULL
 #' }
 #' @export
 setMethod("dbConnect", "MySQLDriver",
-  function(drv, dbname = "", username = "", password = "", host = "",
-    unix.socket = "", port = 0, client.flag = 0,
-    groups = "rs-dbi", default.file = "", ...) {
+  function(drv, dbname = NULL, username = NULL, password = NULL, host = NULL,
+    unix.socket = NULL, port = 0, client.flag = 0,
+    groups = "rs-dbi", default.file = NULL, ...) {
 
     ptr <- connection_create(host, username, password, dbname, port, unix.socket,
       client.flag, groups, default.file)
 
     con <- new("MySQLConnection",
       ptr = ptr,
-      host = host,
-      db = dbname
+      host = ifelse(is.null(host), "", host),
```

Comment:
Not sure what you mean, but I think `NULL` *is* passed down to C++. See the call to `connection_create` in line 57.

Lines 62 and 63 are about what the slots `host` and `db` are filled with *after* the connection has been established. The weird thing is, they are filled with arguments by which `dbConnect()` was called, so from *before* establishing the connection. If those were `NULL` or `""`, for example because they were to be read from an options file, then those empty values will still end up in the slots instead of the actual hostname and database name of the connection as it was made in the end. As far as I can tell, these slots are new since the port to Rcpp, are not used for anything, and their content differs from that of `dbGetInfo()`/`connectionInfo()`, which gathers information about the actually established connection.

So, I also have my doubts about these slots and how they are filled, but I did not want to rewrite everything in a single pull request, so I kept it as I found it for now.

I will change the `ifelse`.